plonk-core/src/proof_system/prover.rs

```diff
@@ -211,20 +217,135 @@ where
+                prover_key.lookup.table_3.clone(),
+                prover_key.lookup.table_4.clone(),
+            ],
+            zeta,
+        );
+
+        // Compute table poly
+        let table_poly = DensePolynomial::from_coefficients_vec(
+            domain.ifft(&compressed_t_multiset.0),
+        );
+
+        // Compute query table f
+        // When q_lookup[i] is zero the wire value is replaced with a dummy
+        //   value currently set as the first row of the public table
+        // If q_lookup[i] is one the wire values are preserved
+        // This ensures the ith element of the compressed query table
+        //   is an element of the compressed lookup table even when
+        //   q_lookup[i] is 0 so the lookup check will pass
+
+        let q_lookup_pad = vec![F::zero(); n - self.cs.q_lookup.len()];
+        let padded_q_lookup =
+            &[self.cs.q_lookup.as_slice(), q_lookup_pad.as_slice()].concat();
+
+        let mut f_scalars: Vec<MultiSet<F>> =
+            vec![MultiSet::with_capacity(w_l_scalar.len()); 4];
+
+        for (q_lookup, w_l, w_r, w_o, w_4) in izip!(
+            padded_q_lookup,
+            w_l_scalar,
+            w_r_scalar,
+            w_o_scalar,
+            w_4_scalar,
+        ) {
+            if q_lookup.is_zero() {
+                f_scalars[0].push(compressed_t_multiset.0[0]);
+                f_scalars.iter_mut().skip(1).for_each(|f| f.push(F::zero()));
+            } else {
+                f_scalars[0].push(*w_l);
+                f_scalars[1].push(*w_r);
+                f_scalars[2].push(*w_o);
+                f_scalars[3].push(*w_4);
+            }
+        }
+
+        // Compress all wires into a single vector
+        let compressed_f_multiset = MultiSet::compress(&f_scalars, zeta);
+
+        // Compute query poly
+        let f_poly = DensePolynomial::from_coefficients_vec(
+            domain.ifft(&compressed_f_multiset.0),
+        );
+
+        // Add blinders to query polynomials
+        // let f_poly = Self::add_blinder(&f_poly, n, 1);
+
+        // Commit to query polynomial
+        let (f_poly_commit, _) =
+            PC::commit(commit_key, &[label_polynomial!(f_poly)], None)
+                .map_err(to_pc_error::<F, PC>)?;
+
+        // Add f_poly commitment to transcript
+        transcript.append(b"f", f_poly_commit[0].commitment());
+
+        // Compute s, as the sorted and concatenated version of f and t
+        let (h_1, h_2) = compressed_t_multiset
+            .sorted_halve(&compressed_f_multiset)
+            .unwrap();
+
+        // Compute h polys
+        let h_1_poly =
+            DensePolynomial::from_coefficients_vec(domain.ifft(&h_1.0));
+        let h_2_poly =
+            DensePolynomial::from_coefficients_vec(domain.ifft(&h_2.0));
+
+        // Add blinders to h polynomials
+        // let h_1_poly = Self::add_blinder(&h_1_poly, n, 1);
+        // let h_2_poly = Self::add_blinder(&h_2_poly, n, 1);
```

Comment:
resolve

Reply:
We will implement plonkup blinding in a separate PR